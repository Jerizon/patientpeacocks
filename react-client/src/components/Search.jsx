import React from 'react';
import Button from 'react-bootstrap/lib/Button';
import ButtonToolbar from 'react-bootstrap/lib/ButtonToolbar'
class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    }
  }


  reqEventful() {
    this.props.eventful();
  }

  reqMeetUP() {
    this.props.meetUp();
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.changeLocation(this.input.value);
  }

  render() {
    return (
      <div>
        <ButtonToolbar>
          <form onSubmit={this.handleSubmit.bind(this)}>
            <label>
              Location:
              <input type="text" ref={(input) => this.input = input} />
            </label>
            <input type="submit" value="Submit" />
          </form>
        </ButtonToolbar>
      </div>
    );
  }
}

export default Search;