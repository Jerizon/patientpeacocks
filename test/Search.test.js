import React from 'react';
import ReactDOM from 'react-dom';
import Search from '../react-client/src/components/Search.jsx';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';


describe('Component: Search', () => {
  
  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <Search />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});