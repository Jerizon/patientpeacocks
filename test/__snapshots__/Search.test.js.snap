// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Search should match its empty snapshot 1`] = `
<div>
  <div
    className="search btn-toolbar"
    role="toolbar"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="search-container"
      >
        Location:
        <input
          placeholder={undefined}
          type="text"
        />
        <input
          className="button"
          type="submit"
          value="Search!"
        />
      </div>
    </form>
  </div>
</div>
`;
